packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -171,24 +154,58 @@ String _artifactToFileName(Artifact artifact, [ TargetPlatform platform, BuildMo
       return 'font-subset$exe';
     case Artifact.constFinder:
       return 'const_finder.dart.snapshot';
-    case Artifact.flutterWebLibrariesJson:
+  }
+  assert(false, 'Invalid artifact $artifact.');
+  return null;
+}
+
+String _hostArtifactToFileName(HostArtifact artifact, bool windows) {
```

Comment:
Or this could be made into an instance method and then `_platform.isWindows` wouldn't have to be passed in at every call site.  Or just keep it consistent with the other artifact methods, either way works.

Reply:
This one is a bit weird because we always implement Artifacts instead of extending, so we'd need to fix a few of the factory constructors. I'm going to leave it as is for now.